includes/rest-api/class-sensei-rest-api-setup-wizard-controller.php

```diff
@@ -0,0 +1,329 @@
+<?php
+/**
+ * Setup Wizard REST API.
+ *
+ * @package Sensei\SetupWizard
+ * @since   3.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Setup Wizard REST API endpoints.
+ *
+ * @package Sensei
+ * @author  Automattic
+ * @since   3.1.0
+ */
+class Sensei_REST_API_Setup_Wizard_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'setup-wizard';
+
+
+	/**
+	 * Main Setup Wizard instance.
+	 *
+	 * @var Sensei_Onboarding
+	 */
+	private $setupwizard;
+
+	/**
+	 * Available 'purpose' options.
+	 */
+	const PURPOSES = [ 'share_knowledge', 'generate_income', 'promote_business', 'provide_certification', 'train_employees', 'other' ];
+
+	/**
+	 * Sensei_REST_API_Setup_Wizard_Controller constructor.
+	 *
+	 * @param string $namespace Routes namespace.
+	 */
+	public function __construct( $namespace ) {
+		$this->namespace   = $namespace;
+		$this->setupwizard = Sensei_Onboarding::instance();
+	}
+
+	/**
+	 * Register the REST API endpoints for Setup Wizard.
+	 */
+	public function register_routes() {
+
+		$this->register_progress_route();
+		$this->register_welcome_route();
+		$this->register_purpose_route();
+		$this->register_features_route();
+	}
+
+	/**
+	 * Check user permission for REST API access.
+	 *
+	 * @return bool Whether the user can access the Setup Wizard REST API.
+	 */
+	public function can_user_access_rest_api() {
+		return current_user_can( 'manage_sensei' );
+	}
+
+	/**
+	 * Register /welcome endpoint.
+	 */
+	public function register_welcome_route() {
```

Comment:
Would it make sense to you rename these methods to `register_edit_..._route`? So it identifies that the methods are for edit purposes.

Reply:
👍  7cdd3c5e